test_util/src/lib.rs

```diff
@@ -857,7 +880,9 @@ async fn main_server(req: Request<Body>) -> hyper::Result<Response<Body>> {
         return Ok(file_resp);
       }
 
-      return Ok(Response::new(Body::empty()));
+      Response::builder()
```

Comment:
Strangly, the test server returned an empty body irrespective of the URL requested, instead of a 404 not found.  While this was specific to creating the "sniffing" behaviour reverted, it still feels like we should return a 404 like a "normal" webserver to ensure the code behaves properly.